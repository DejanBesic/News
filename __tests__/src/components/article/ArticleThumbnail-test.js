import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import { ArticleThumbnail } from 'Components';
import { TouchableWithoutFeedback } from 'react-native-gesture-handler';

describe('ArticleThumbnail', () => {
  let props = {};
  const mock = jest.fn();
  beforeAll(() => {
    props = {
      title: 'Title',
      description: 'Description',
      onPress: mock,
      urlToImage: 'https://fakeurl.com/someimageurl.jpg',
    };
  });

  it('doesn\t crash without props', () => {
    const tree = renderer.create(<ArticleThumbnail onPress={mock} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    const tree = renderer.create(<ArticleThumbnail {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('triggers onPress correctly', () => {
    const tree = renderer.create(<ArticleThumbnail {...props} />);
    const container = tree.root.findByType(TouchableWithoutFeedback);
    renderer.act(() => {
      container.props.onPress();
    });
    expect(mock).toHaveBeenCalled();
  });
});
