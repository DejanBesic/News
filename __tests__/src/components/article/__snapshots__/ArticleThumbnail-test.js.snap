// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleThumbnail doesn	 crash without props 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
>
  <View
    accessible={true}
    style={
      Object {
        "backgroundColor": "rgba(218,211,200,0.7)",
        "borderRadius": 5,
        "paddingBottom": 15,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 15,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "lineHeight": 20,
          },
        ]
      }
    >
      
    </Text>
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "height": 200,
              "marginBottom": 5,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 5,
            },
          ],
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "priority": "normal",
            "uri": "",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
          },
        ]
      }
    >
      
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "display": "flex",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2f92dd",
              "fontSize": 15,
            },
          ]
        }
      >
        More &gt;
      </Text>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`ArticleThumbnail renders correctly with props 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
>
  <View
    accessible={true}
    style={
      Object {
        "backgroundColor": "rgba(218,211,200,0.7)",
        "borderRadius": 5,
        "paddingBottom": 15,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 15,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "lineHeight": 20,
          },
        ]
      }
    >
      Title
    </Text>
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "height": 200,
              "marginBottom": 5,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 5,
            },
          ],
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "priority": "normal",
            "uri": "https://fakeurl.com/someimageurl.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
          },
        ]
      }
    >
      Description
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "display": "flex",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2f92dd",
              "fontSize": 15,
            },
          ]
        }
      >
        More &gt;
      </Text>
    </View>
  </View>
</RNGestureHandlerButton>
`;
